OpenVote/src/com/openvote/LoginServlet.java

```diff
@@ -41,7 +51,29 @@ public void doPost(HttpServletRequest req, HttpServletResponse resp)
 		{
 			e.printStackTrace();
 		}
+	
+ 
+		} else 
+   		 if (user != null){
+   		resp.sendRedirect(userService.createLogoutURL("/login.jsp"));
```

Comment:
There are three cases:
1. there's a user logged in, and that user is the admin.
2. there's a user logged in, and that user is NOT the admin
3. no user logged in

Line 58, which you've highlighted, takes care of case 2 and logs out the user (creating case 3)

Line 62 takes care of case 3.